flycheck-rust.el

```diff
@@ -69,6 +69,14 @@ Return the directory containing the Cargo file, or nil if there
 is none."
   (locate-dominating-file (buffer-file-name) "Cargo.toml"))
 
+(defun flycheck-rust-find-crate-root ()
+  "Get the crate root (the nearest lib.rs or main.rs)
+relative to the current file."
+  (-if-let (lib-crate-dir (locate-dominating-file (buffer-file-name) "lib.rs"))
+      (concat lib-crate-dir "lib.rs")
```

Comment:
Please use `expand-file-name` to join paths, i.e. `(expand-file-name "lib.rs" lib-crate-dir)`